pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +135,31 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			if err := os.MkdirAll(hostPath, perm); err != nil {
+				glog.Errorf("failed to mkdir:%s", hostPath)
+				return nil, err
+			}
+
+			// If umask is preventing us from making the sub path have the same permissions as the
+			// mounter path, chmod the sub path
+			fileinfo, err = os.Lstat(hostPath)
```

Comment:
why bother to do an lstat() at all?  Just set the mode - same number of syscalls in the best case, and less is the worst case.